sympy/physics/mechanics/kane.py

```diff
@@ -581,7 +622,10 @@ def kanes_equations(self, bodies=None, loads=None):
                 km = KanesMethod(self._inertial, self.q, self._uaux,
                         u_auxiliary=self._uaux, u_dependent=self._udep,
                         velocity_constraints=(self._k_nh * self.u +
-                        self._f_nh))
+                        self._f_nh),
+                        acceleration_constraints=(self._k_dnh * self._udot +
```

Comment:
Not clear why this is added.

Reply:
if this is not done, then the 2nd call to KanesMethod ends up having to recompute the acceleration constraint. While in principle that's fine, it's an issue if acceleration_constraints were passed to the initial call: the initial call uses line 228 to compute `_f_dnh` and `_k_dnh` from `acc`, while the 2nd call uses line 220 to compute them from `vel` 
By passing `acceleration_constraints = self._k_dnh * self._udot +self._f_dnh`, it forces the 2nd call to use 228 which will just recover `_f_dnh` and `_k_dnh` which have been computed in the initial call regardless of whether acceleration constraints was passed or not.

Why is this relevant to this explicit kinematics PR? acceleration_constraints might need to be passed to ensure that no qdots appear in them, and not rely on KanesMethod computing them from derivatives of velocity constraints which might introduce qdots.